### PR TITLE
784 fix ac dc powers

### PR DESCRIPTION
**Pull request by brtietz**
Define additional variables and conventions in the powerflow variables according to https://github.com/NREL/ssc/issues/784. Closes https://github.com/NREL/ssc/issues/784

Conventions for DC to AC conversion for AC connected batteries and DC connected batteries while discharging are preserved. DC connected batteries charging from PV needed new a convention, which is:

Convert to AC power by multiplying by the inverter max efficiency.

This PR also updates some of the losses conventions to ensure "electricity from grid to system" matches the inverter AC "output" when charging. 

Alternative conventions considered included "what would it take to charge from the grid (divide by inverter efficiency)" and "use actual inverter efficiency". Some notes on what didn't work:
* We need AC variables for LCOS to make sense (Cannot report DC only)
* Both "electricity to/from grid" and "from system to battery" need to use the same convention, otherwise "percent charge from system" doesn't work
*  The grid charging convention is problematic because it makes the DC connected battery look less efficient than the AC connected battery
* Using the actual inverter efficiency could push the round-trip efficiency even higher than the current convention

Additional changes:
- Use night time losses when deciding whether or not the inverter is running in the powerflow code, rather than assuming a 5% efficiency
- Modifications to dispatch code to be more consistent about AC vs DC power in the constraints
- Solve a few compiler warnings

Relevant test files:
Ensure grid charging and PV charging numbers add up for a DC connected battery: 
[dc_charging_grid_and_pv.zip](https://github.com/NREL/ssc/files/9847457/dc_charging_grid_and_pv.zip)
Custom dispatch from pvsamv1_battery integration tests that produces low inverter efficiencies: 
[low_power_inverter_test_cases.zip](https://github.com/NREL/ssc/files/9847458/low_power_inverter_test_cases.zip)

Looking for at least two approvals prior to merge.

I expect there will be some changes to the SAM_api and export_config files, but those builds are broken on develop at to time of writing (the relevant person has been emailed). I'll make a SAM PR for that when they're fixed.

### CODE REVIEW

**Comment by cpaulgilman** [comment]
@brtietz Thanks for going through this, and for the clear descriptions in your comment above.

The relationship between the DC and AC versions of "Electricity to battery from system" is the inverter's peak nominal efficiency, For a DC-connected battery, the calculated AC value is an approximation for LCOS calculations and not a power quantity that you would measure in a real system. Is that correct?

**Review by cpaulgilman — APPROVED** [review]

**Comment by brtietz** [comment]
> @brtietz Thanks for going through this, and for the clear descriptions in your comment above.
> 
> The relationship between the DC and AC versions of "Electricity to battery from system" is the inverter's peak nominal efficiency, For a DC-connected battery, the calculated AC value is an approximation for LCOS calculations and not a power quantity that you would measure in a real system. Is that correct?

For PV charging yes. For grid charging, it is the power draw on the AC side of the inverter.

File: shared/lib_battery_dispatch.cpp [hunk 1]
```diff
@@ -174,7 +174,7 @@ bool dispatch_t::check_constraints(double& I, size_t count)
     // Don't allow grid charging unless explicitly allowed (reduce charging)
     if (!m_batteryPower->canGridCharge && I < 0 && m_batteryPower->powerGridToBattery > tolerance)
     {
-        m_batteryPower->powerBatteryTarget += m_batteryPower->powerGridToBattery;
+        m_batteryPower->powerBatteryTarget += m_batteryPower->powerGridToBattery * m_batteryPower->singlePointEfficiencyACToDC;
```
Comment (sjanzou) [inline shared/lib_battery_dispatch.cpp:177]: Are we going to handle different efficiencies in a future issue or pull request?

**Review by sjanzou — APPROVED** [review]
Lots of clarifying outputs and nice use of the code-attic!

File: shared/lib_battery_dispatch.cpp (hunk 1 above)
Comment (brtietz) [reply shared/lib_battery_dispatch.cpp:177]: Yes, if we do some of the DC-DC converter work we talked about with the PVSAMV1 refactor. Some of the DC-DC conversions are currently handled early in the powerflow code, the current setup should work as long as we keep using single point efficiencies.

**Pull request merged**